src/app/[locale]/(default)/storage/page.tsx

```diff
@@ -0,0 +1,158 @@
+import { useTranslations } from 'next-intl';
+import { getTranslations, unstable_setRequestLocale } from 'next-intl/server';
+import Image from 'next/image';
+import { createSearchParamsCache, parseAsInteger } from 'nuqs/parsers';
+
+import { PaginationCarousel } from '@/components/layout/PaginationCarousel';
+import { Button } from '@/components/ui/Button';
+import {
+  Card,
+  CardContent,
+  CardDescription,
+  CardFooter,
+  CardHeader,
+  CardTitle,
+} from '@/components/ui/Card';
+import { Combobox } from '@/components/ui/Combobox';
+import { SearchBar } from '@/components/ui/SearchBar';
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from '@/components/ui/Select';
+
+export async function generateMetadata({
+  params: { locale },
+}: {
+  params: { locale: string };
+}) {
+  const t = await getTranslations({ locale, namespace: 'layout' });
+
+  return {
+    title: t('storage'),
+  };
+}
+
+export default function StoragePage({
+  params: { locale },
+  searchParams,
+}: {
+  params: { locale: string };
+  searchParams: Record<string, string | string[] | undefined>;
+}) {
+  unstable_setRequestLocale(locale);
+  const t = useTranslations('storage');
+  const t_ui = useTranslations('ui');
+
+  const itemsPerPage = 12;
+
+  const searchParamsCache = createSearchParamsCache({
+    [t_ui('page')]: parseAsInteger.withDefault(1),
+  });
+
+  const { [t_ui('page')]: page = 1 } = searchParamsCache.parse(searchParams);
+
+  // TODO: Implement filters and category selection
+  const categories = [
+    {
+      value: 'cables',
+      label: t('combobox.cables'),
+    },
+    {
+      value: 'sensors',
+      label: t('combobox.sensors'),
+    },
+    {
+      value: 'peripherals',
+      label: t('combobox.peripherals'),
+    },
+    {
+      value: 'miniPC',
+      label: t('combobox.miniPC'),
+    },
+  ];
+
+  const filters = [
+    'select.popularity',
+    'select.sortDescending',
+    'select.sortAscending',
+    'select.name',
+  ] as const;
+
+  return (
+    <>
+      <div className='my-4 flex flex-col justify-center gap-2 lg:flex-row'>
+        <SearchBar className='lg:max-w-2xl' />
+        <Select>
+          <SelectTrigger className='w-full lg:w-[250px]'>
+            <SelectValue placeholder={t('select.defaultPlaceholder')} />
+          </SelectTrigger>
+          <SelectContent>
+            {filters.map((filter) => (
+              <SelectItem key={filter} value={filter}>
+                {t(filter)}
+              </SelectItem>
+            ))}
+          </SelectContent>
+        </Select>
+
+        <Combobox
+          choices={categories}
+          defaultDescription={t('combobox.defaultDescription')}
+          defaultPlaceholder={t('combobox.defaultPlaceholder')}
+          buttonClassName='w-full lg:w-[250px]'
+          contentClassName='w-full lg:w-[200px]'
+        />
+      </div>
+      <div className='grid grid-cols-1 gap-3 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4'>
```

Comment:
Maybe it looks better to switch from sm to xs so it keeps two columns until the screen is even smaller

Reply:
Yeah, it does indeed look better. Will get it changed